ostd/src/io_mem.rs

```diff
@@ -34,16 +50,45 @@ impl IoMem {
     /// - The given physical address range must be in the I/O memory region.
     /// - Reading from or writing to I/O memory regions may have side effects. Those side effects
     ///   must not cause soundness problems (e.g., they must not corrupt the kernel memory).
-    pub(crate) unsafe fn new(range: Range<Paddr>) -> Self {
+    pub(crate) fn new(range: Range<Paddr>) -> Self {
```

Comment:
The function should still be `unsafe`, because that `IoMem` provides `VmIo`, the provided physical address will be accessible from the created `IoMem`. The caller should ensure that the physical address does not overlap with tracked memory or regions that are not I/O memory.